### ISSUE TITLE
Add graph traversal algorithms

### ISSUE THREAD

**Issue by faheel** [ProAlgos/ProAlgos-Cpp#11]
Create a folder named `Graph` in the root directory. Inside it, the following graph traversal algorithms are to be added:
- [Breadth-first search](https://en.wikipedia.org/wiki/Breadth-first_search) (in `Traversal/BreadthFirstSearch.cpp`)
- [Depth-first search](https://en.wikipedia.org/wiki/Depth-first_search) (in `Traversal/DepthFirstSearch.cpp`)

Use either an _adjacency list_ (recommended) or an _adjacency matrix_ to implement an undirected weighted [graph](https://en.wikipedia.org/wiki/Graph_(abstract_data_type)).

Use STL containers such as `std::queue` and `std::stack` for implementing a queue and stack respectively.

Create a header file named `Graph.h` (in `DataStructures/Graph`) and implement the basic functions for getting inputs from the user, such as:
- number of vertices
- number of edges
- starting vertex
- edge info (source, destination and weight)

**Comment by Dehorser** [comment]
I'd love to take this one.

**Comment by Dehorser** [comment]
How should I represent the vertices? I'm currently using a struct named "vertex."

**Comment by Dehorser** [comment]
Just kidding. I notice another user has taken this job.

**Comment by faheel** [comment]
### Note
As I mentioned in issue #70, I will be adding 4 classes for directed and undirected graphs using adjacency matrix and adjacency list representations, but only after issue #74 is taken care of.
Once the classes are added, anyone interested in taking this up, or other graph algorithms, can send their PRs. Please wait until then.

**Comment by faheel** [comment]
Issue #74 is now closed

**Comment by faheel** [comment]
Closing this since there haven't been any updates for a long time. If you'd like to work on graph traversal algorithms, please open a new issue for each such algorithm.
